src/Haskell/Ide/Engine/Support/HieExtras.hs

```diff
@@ -799,3 +801,12 @@ prefixes =
   , "$c"
   , "$m"
   ]
+
+-- ---------------------------------------------------------------------
+
+getFormattingPlugin :: Config -> IdePlugins -> Maybe (PluginDescriptor, FormattingProvider)
+getFormattingPlugin config plugins = do 
+  let providerName = formattingProvider config
+  fmtPlugin <- Map.lookup providerName (ipMap plugins)
```

Comment:
This is actually wrong, since we look up the name, but the Plugins Map is of Type `Map PluginId PluginDescriptor`

Reply:
But it works, since due to some lucky coincidence `pluginId == pluginName`.

Reply:
I think it would be better if PluginId would be a `newtype`.